internal/activity/database/activity_repository.go

```diff
@@ -0,0 +1,122 @@
+package database
+
+import (
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/sumelms/microservice-activity/internal/activity/domain"
+	"github.com/sumelms/microservice-activity/pkg/errors"
+)
+
+// NewActivityRepository creates the activity ActivityRepository.
+func NewActivityRepository(db *sqlx.DB) (ActivityRepository, error) { //nolint: revive
+	sqlStatements := make(map[string]*sqlx.Stmt)
+
+	for queryName, query := range queriesActivity() {
+		stmt, err := db.Preparex(query)
+		if err != nil {
+			return ActivityRepository{},
+				errors.WrapErrorf(err, errors.ErrCodeUnknown, "error preparing statement %s", queryName)
+		}
+		sqlStatements[queryName] = stmt
+	}
+
+	return ActivityRepository{
+		statements: sqlStatements,
+	}, nil
+}
+
+type ActivityRepository struct {
+	statements map[string]*sqlx.Stmt
+}
+
+func (r ActivityRepository) statement(s string) (*sqlx.Stmt, error) {
+	stmt, ok := r.statements[s]
+	if !ok {
+		return nil, errors.NewErrorf(errors.ErrCodeUnknown, "prepared statement %s not found", s)
+	}
+	return stmt, nil
+}
+
+func (r ActivityRepository) Activity(id uuid.UUID) (domain.Activity, error) {
```

Comment:
We have adopted the pattern of having the parameters: `id  uuid.UUID` and are filling with the property `uuid`. 

With the aim of making it clearer what we are doing: _Should we rename this parameters to `uuid  uuid.UUID` or use the `id` property to fill these parameters?_

@arianerocha @rluders

Reply:
In this case, if you use `uuid uuid.UUID` you can easily shadow the `uuid` package import. So, I'm not against changing it, it would be even easier to read and understand but keep in mind that you will have to create an alias for the uuid package import.

Reply:
I think we can adopt `${field}_ uuid` so when someone is using our functions will be clear with uuid we expect

Reply:
In this case, you can change the method signature for:

```go
func (r ActivityRepository) Activity(activityUUID uuid.UUID) (domain.Activity, error) {
```

or

```go
func (r ActivityRepository) Activity(uid uuid.UUID) (domain.Activity, error) {
```